src/components/ButtonSystem/MainButtons.style.tsx

```diff
@@ -0,0 +1,92 @@
+import styled, { css } from 'styled-components';
+import { ButtonType, ButtonsTheme } from 'assets/theme/buttons';
+import BaseButton from './BaseButton';
+
+function button(styles: ButtonsTheme) {
+  return css`
+    text-decoration: none;
+    text-transform: none;
+    border-radius: 4px;
+    line-height: 1;
+    cursor: pointer;
+    letter-spacing: 0;
+    ${styles.fontFamily};
+    background-color: ${styles.background};
+    color: ${styles.text};
+    border: 1px solid ${styles.border};
+    transition: background-color 0.1s linear;
+
+    &:hover {
+      background-color: ${styles.backgroundHover};
+      color: ${styles.textHover};
+      border: 1px solid ${styles.borderHover};
+    }
+
+    svg {
+      color: ${styles.icon};
+    }
+
+    &:disabled {
+      background-color: ${styles.disabledBackground};
+      color: ${styles.disabledText};
+
+      svg {
+        color: ${styles.disabledIcon};
+      }
+    }
+  `;
+}
+
+/**
+ * max-height makes sure buttons with + without icons are the same height,
+ * since MUI icons have built-in padding that we are accounting for
+ */
+const standardButtonSizing = css`
+  padding: 0.625rem;
+  font-size: 0.875rem;
+  max-height: 2.25rem;
+`;
+
+const largeButtonSizing = css`
+  padding: 1rem;
+  font-size: 1rem;
+  max-height: 3.125rem;
+`;
+
+const textButtonSizing = css`
+  padding: 0.625rem;
+  font-size: 1rem;
+  max-height: 2.25rem;
+
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+
+/**
+ * Main button variations:
+ */
+export const OutlinedButton = styled(BaseButton)`
```

Comment:
This is a small drive by comment, but I wonder if it would be worth having these different variants accessible by either a prop (similar to material ui variant field `variant="filled"`, `variant="outlined"`, etc.) or something like: 
```
export {
  Outlined: OutlinedButton,
  Filled: FilledButton
}
```
so you can call something similar to
```
Buttons.Outlined
```
would be useful.

I think having all the different variants makes the API feel more streamlined but may just be a personal preference!

Reply:
in terms of using a prop-- i thought about this, but i landed on the current setup so that we dont need to import 2 things (a button + an enum for the prop) when implementing, and we could just import 1 thing (an intuitively-named button component)

Reply:
Ooo the double import thing makes sense.  i think material ui handles that a bit differently by exporting a type, something like `export type FormControlClassKey = 'root' | 'marginNormal' | 'marginDense' | 'fullWidth';`

Then the prop is typed as that type, but can define it easily as a string.  (also i went down a rabbit hole looking into this because i've kinda always been curious how they do it, but they just [make a map of variant -> component](https://github.com/mui-org/material-ui/blob/17fe2c49d1cd081036c549ada51f49e0061b2923/packages/material-ui/src/TextField/TextField.js#L129) and return that - kinda cool!)

Reply:
ooo handy + good to know!! gonna keep this pr as is